Collections/Queue.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Collections
+{
+    internal static class Queue
+    {
+
+        public static int GetMaxValue(Queue<int> queue)
+        {
+            if (queue.Count == 0)
+            {
+                throw new InvalidOperationException("Queue is empty.");
+            }
+            int[] inputs = queue.ToArray();
```

Comment:
The tricky move is to convert the queue into an array :smirk:
But you need to work only with queue and practice using methods such as Peek, Enqueue, Dequeue))